addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/UniFiController.java

```diff
@@ -0,0 +1,403 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.unifi.internal.api.UniFiClient;
+import org.openhab.binding.unifi.internal.api.UniFiDevice;
+import org.openhab.binding.unifi.internal.api.UniFiSite;
+import org.openhab.binding.unifi.internal.api.UniFiWiredClient;
+import org.openhab.binding.unifi.internal.api.UniFiWirelessClient;
+import org.openhab.binding.unifi.internal.api.json.UniFiClientResponse;
+import org.openhab.binding.unifi.internal.api.json.UniFiDeviceResponse;
+import org.openhab.binding.unifi.internal.api.json.UniFiSiteResponse;
+import org.openhab.binding.unifi.internal.ssl.UniFiHostnameVerifier;
+import org.openhab.binding.unifi.internal.ssl.UniFiTrustManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ * The {@link UniFiController} is the main communication point with an external instance of the Ubiquiti Networks
+ * Controller Software.
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+public class UniFiController {
+
+    private static final Map<String, String> INSIGHTS_WITHIN_24H = Collections.singletonMap("within", "24");
+
+    private Logger logger = LoggerFactory.getLogger(UniFiController.class);
+
+    private UniFiControllerConfig config;
+
+    private HostnameVerifier hostnameVerifier = new UniFiHostnameVerifier();
+
+    private SSLSocketFactory sslSocketFactory;
+
+    private Map<String, UniFiSite> sitesCache = Collections.emptyMap();
+
+    private Map<String, UniFiDevice> devicesCache = Collections.emptyMap();
+
+    private Map<String, UniFiClient> clientsCache = Collections.emptyMap();
+
+    private Map<String, UniFiClient> insightsCache = Collections.emptyMap();
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private CookieManager cookieManager = new CookieManager();
+
+    public UniFiController(UniFiControllerConfig config) throws UniFiException {
+        try {
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, new TrustManager[] { new UniFiTrustManager() }, new SecureRandom());
+            sslSocketFactory = sc.getSocketFactory();
+        } catch (Exception e) {
+            throw new UniFiException("Could not install SSL trust-all manager", e);
+        }
+        objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
+        if (!config.isValid()) {
+            throw new UniFiException("Invalid configuration");
+        } else if (config.getConsiderHome().compareTo(config.getRefresh()) <= 0) {
```

Comment:
See comment in `UniFiControllerConfig`.
This expression will be simplified when the `BigDecimal` configuration values are replaced with `int` values. :-)